### PR TITLE
feat: create main chat logic

### PR DESCRIPTION
**Pull request by OlgaSavina**
## Describe your changes
- I have done main logic for chats. I have added main endpoins for creating rooms(chats) and messages. Sockets will be added on next prs
## Issue ticket code (and/or) and link
- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?selectedIssue=AG-209)
photo_2023-12-26_20-02-34

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [ ] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [ ] use ConfigService instead of process.env
- [ ] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

**Review by rudenkoserhii — APPROVED** [review]

File: src/common/entities/message.entity.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { User } from './user.entity';
+import { Room } from './room.entity';
+import { CoreEntity } from './core.entity';
+
+@Entity()
+export class Message extends CoreEntity {
+ 
+
```
Comment (lyudad) [inline src/common/entities/message.entity.ts]: empty line not needed here, I think you have issue with pritier

File: src/common/entities/room.entity.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { User } from './user.entity';
+import { CoreEntity } from './core.entity';
+
+  @Entity()
+  export class Room extends CoreEntity {
+   
+    @Column()
+    title: string;
+  
+
```
Comment (lyudad) [inline src/common/entities/room.entity.ts]: same

File: src/modules/room/dto/create-room.dto.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      })
+  @IsNotEmpty()
+  @MinLength(1)
+  @MaxLength(100)
+  title: string;
+
+ 
+
+
+
```
Comment (lyudad) [inline src/modules/room/dto/create-room.dto.ts]: formatting issue

File: src/modules/message/message.controller.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  Request,
+} from '@nestjs/common';
+
+import { CreateMessageDto } from './dto/create-message.dto';
+import { GetMessagesDto } from './dto/get-messages.dto';
+import { MessageService } from './message.service';
+import { JwtAuthGuard } from '../auth/guards/jwt-auth.guard';
+import { ApiBearerAuth, ApiOperation, ApiResponse } from '@nestjs/swagger';
+import { Message } from 'src/common/entities/message.entity';
+
```
Comment (AKutusenko) [inline src/modules/message/message.controller.ts:8]: порядок импортов

File: src/modules/message/message.module.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Module } from '@nestjs/common';
+import { TypeOrmModule } from '@nestjs/typeorm';
+
+import { MessageController } from './message.controller';
+
+import { MessageService } from './message.service';
+import { Message } from 'src/common/entities/message.entity';
+import { RoomModule } from '../room/room.module';
+
```
Comment (AKutusenko) [inline src/modules/message/message.module.ts:8]: тоже

File: src/modules/message/message.service.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Message } from 'src/common/entities/message.entity';
+import { Repository } from 'typeorm';
+import { GetMessagesDto } from './dto/get-messages.dto';
+import { CreateMessageDto } from './dto/create-message.dto';
+
```
Comment (AKutusenko) [inline src/modules/message/message.service.ts:7]: тоже

**Review by AKutusenko — APPROVED** [review]

File: src/modules/message/message.controller.ts (hunk 4 above)
Comment (OlgaSavina) [reply src/modules/message/message.controller.ts:8]: fixed

File: src/modules/message/message.module.ts (hunk 5 above)
Comment (OlgaSavina) [reply src/modules/message/message.module.ts:8]: same

**Pull request merged**